lib/gui/components/drive-selector/controllers/drive-selector.js

```diff
@@ -142,4 +142,34 @@ module.exports = function($q, $uibModalInstance, DrivesModel, SelectionStateMode
     });
   };
 
+  this.getButtonStatus = () => {
+    const isDisabled = !this.state.hasDrive() || this.state.getDrive().protected;
+
+    if (isDisabled) {
+      return 'disabled';
+    }
+
+    const isDangerous = this.state.hasDrive() && this.state.isSystemDrive(this.state.getDrive());
+
+    if (isDangerous) {
+      return 'danger';
+    }
+
+    return 'primary';
+  };
+
+  this.getFooterMessage = () => {
+    if (this.getButtonStatus() === 'danger'
+      && this.state.isSystemDrive(this.state.getDrive())) {
```

Comment:
Based on:

```
const isDangerous = this.state.hasDrive() && this.state.isSystemDrive(this.state.getDrive());
```

Then if the button status is `danger`, then it follows that the drive is a system drive, therefore I think this conditional could be reduced to:

```
if (this.getButtonStatus() === 'danger') {}
```